src/perl/Permabit/BlockDevice/ISCSI.pm

```diff
@@ -238,6 +225,37 @@ sub getDevicePath {
   return $self->SUPER::getDevicePath();
 }
 
+########################################################################
+# @inherit
+##
+sub addToDevicesFile {
+  my ($self) = assertNumArgs(1, @_);
+  if ($self->isPassThrough()) {
+    # Pass through doesn't actually exist
+    return;
+  }
+  my $lsscsiCmd = "lsscsi 2>/dev/null | grep pbit-target";
+  my $lsscsi = $self->runOnHostIgnoreErrors($lsscsiCmd);
+  my $device = (split('\s', $lsscsi))[-1];
+  $self->runOnHost("sudo lvmdevices -y --adddev $device");
+}
+
+########################################################################
+# @inherit
+##
+
+sub removeFromDevicesFile {
+  my ($self) = assertNumArgs(1, @_);
+  if ($self->isPassThrough()) {
+    # Pass through doesn't actually exist
+    return;
+  }
+  my $lsscsiCmd = "lsscsi 2>/dev/null | grep pbit-target";
+  my $lsscsi = $self->runOnHostIgnoreErrors($lsscsiCmd);
+  my $device = (split('\s', $lsscsi))[-1];
+  $self->runOnHost("sudo lvmdevices -y --deldev $device");
+}
+
 ########################################################################
```

Comment:
This creates the possibility of a failure mode that didn't exist before: The remove...() function might get a different value for the device name than the add...() function used. We could get around that by (for instance) caching the device name in the ISCSI object.

I'm not sure this failure mode can actually happen, though, at least as long as we're still on the same machine where he evice was added in the first place. (If we're not, that leads to other problems.)

Reply:
Its strange. I had this the way you want it and then I ran the ISCSIDevice01 test and it failed, so I did it this way because of how the device was changing name during migration. But maybe I screwed something up. I'll revert and try the test again.

Reply:
I think I've come up with a solution for this. I've moved the adddeactivation step into addToDevicesFile where it can properly use the right device name.